src/components/Attachments/AttachmentCarousel/CarouselItem.js

```diff
@@ -0,0 +1,117 @@
+import React, {useContext, useState} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import CONST from '../../../CONST';
+import styles from '../../../styles/styles';
+import useLocalize from '../../../hooks/useLocalize';
+import PressableWithoutFeedback from '../../Pressable/PressableWithoutFeedback';
+import Text from '../../Text';
+import Button from '../../Button';
+import AttachmentView from '../AttachmentView';
+import SafeAreaConsumer from '../../SafeAreaConsumer';
+import ReportAttachmentsContext from '../../../pages/home/report/ReportAttachmentsContext';
+
+const propTypes = {
+    /** Attachment required information such as the source and file name */
+    item: PropTypes.shape({
+        /** Report action ID of the attachment */
+        reportActionID: PropTypes.string,
+
+        /** Whether source URL requires authentication */
+        isAuthTokenRequired: PropTypes.bool,
+
+        /** The source (URL) of the attachment */
+        source: PropTypes.string,
+
+        /** File additional information of the attachment */
+        file: PropTypes.shape({
+            /** File name of the attachment */
+            name: PropTypes.string,
+        }),
+
+        /** Whether the attachment has been flagged */
+        hasBeenFlagged: PropTypes.bool,
+    }).isRequired,
+
+    /** Whether the attachment is currently being viewed in the carousel */
+    isFocused: PropTypes.bool.isRequired,
+
+    /** onPress callback */
+    onPress: PropTypes.func,
+};
+
+const defaultProps = {
+    onPress: undefined,
+};
+
+function CarouselItem({item, isFocused, onPress}) {
+    const {translate} = useLocalize();
+    const {isAttachmentHidden} = useContext(ReportAttachmentsContext);
+    const [isHidden, setIsHidden] = useState(() => {
+        const isAttachmentHiddenValue = isAttachmentHidden(item.reportActionID);
+        return isAttachmentHiddenValue === undefined ? item.hasBeenFlagged : isAttachmentHiddenValue;
```

Comment:
NAB but I'd prefer `isAttachmentHidden`

Reply:
Let's go for it, I think that improves readability considerably and is valid given we're supporting it soon.